src/bin/cli.rs

```diff
@@ -386,3 +402,70 @@ fn lua_kube_list_stub<'lua>(
             .serialize_unit_to_null(false),
     )
 }
+
+async fn cli_check(args: CheckArgs) -> Result<()> {
+    for cronpolicy_path in args.cron_policy_paths {
+        let cronpolicy_path_span =
+            tracing::info_span!("cronpolicy-file", path = %cronpolicy_path.display());
+        check_cronpolicy_path(&cronpolicy_path)
+            .instrument(cronpolicy_path_span)
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to check for cronpolicy file `{}`",
+                    cronpolicy_path.display()
+                )
+            })?;
+    }
+    Ok(())
+}
+
+async fn check_cronpolicy_path(cronpolicy_path: &Path) -> Result<()> {
+    // Open and deserialize cronpolicy file
+    let cronpolicy_file =
+        fs::File::open(cronpolicy_path).context("failed to open cronpolicy file")?;
```

Comment:
nit: also call `fs` operation asynchronously?

Reply:
`serde_yaml::from_reader` only takes `std::io::Read`, not `AsyncRead`. So we have to either asynchronously read all file to memory and then parse, or synchronously read and parse without reading the entire file into memory.